File: BACKEND/Shared/Dtos/StopDto.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Shared.Dtos
{
    public class StopRequest
    {
        [Required]
        public required string Name { get; set; }
        public  string? Address { get; set; }
        public  string? Description { get; set; }

        [Required]
        public required decimal x { get; set; }
        [Required]
        public required decimal y { get; set; }
        public string? Status { get; set; }
      
    }

    public class StopResponse
    {
        public int Id { get; set; }
        public required string Name { get; set; }
        public string? Address { get; set; }
        public string? Description { get; set; }
        public required decimal x { get; set; }
        public required decimal y { get; set; }
        public string? Status { get; set; }
        
    }
}

File: FRONTEND/src/app/shared/types/Dtos/stop.dto.ts

export interface StopRequest{
    name:string;
    address:string;
    x:number;
    description:string;
    y:number;
    status:string;
}


export interface StopResponse{
    id:number;
    name:string;
    address:string;
    description:string;
    x:number;
    y:number;
    point:number[];
    status:string;
}